### PR TITLE
Bug extraction

### PR DESCRIPTION
**Pull request by debora-pe**
This is to fix a small bug that @marlageha encountered.

In `core/skysub.py/local_skysub_extract` during the profile fitting iterations, it can happen that `sobjs[iobj].OPT_MASK` has only one good pixel and this will make `extract.fit_profile` to crash.

In this PR I added a condition for which  `sobjs[iobj].OPT_MASK` must have at least 5 good pixels to update the extraction.

I run the tests and everything passes except the known problem with `shane_kast_red`

```
--- PYPEIT DEVELOPMENT SUITE FAILED 2/128 TESTS (Masters ignored) ---
Failed tests:
    shane_kast_red/600_7500_d57 pypeit (without masters)
    shane_kast_red/1200_5000_d57 pypeit (without masters)
Skipped tests:
Testing Started at 2021-08-10T03:05:36.643918
Testing Completed at 2021-08-10T10:24:17.412914
Total Time: 7:18:40.768996
```

`========================== 287 passed in 1185.32s (0:19:45) ==============================`

### CODE REVIEW

**Review by profxj — APPROVED** [review]
I think this explains a crash I am seeing in QL too!

Thx

File: pypeit/core/skysub.py [hunk 1]
```diff
@@ -735,7 +735,9 @@ def local_skysub_extract(sciimg, sciivar, tilts, waveimg, global_sky, rn2_img,
                                             last_profile, box_rad, sobjs[iobj])
                     # If the extraction is bad do not update
                     if sobjs[iobj].OPT_MASK is not None:
-                        if sobjs[iobj].OPT_MASK.any():
+                        # if sobjs[iobj].OPT_MASK.any():
+                        # if there is only one good pixel `extract.fit_profile` fails
+                        if np.where(sobjs[iobj].OPT_MASK == True)[0].size > 5:
```
Comment (jhennawi) [inline pypeit/core/skysub.py]: Please make this:

if np.sum(sobjs[iobj].OPT_MASK) > extract_good_frac*nspec: 
   etc. 

This removes the arbitrariness of the number 5. Please add extract_good_frac as  keyword parameter in local_skysub_extract, document it, and set the default to what you think an appropriate number is, which is I guess is something like 0.01.

Comment (jhennawi) [reply pypeit/core/skysub.py]: I should add, don't bother to add this to the parset, just have it be a keyword argument for now.

**Review by jhennawi — CHANGES_REQUESTED** [review]
Minor change requested.

File: pypeit/core/skysub.py (hunk 1 above)
Comment (debora-pe) [reply pypeit/core/skysub.py]: Thanks for the feedback @jhennawi !
I made the changes that you suggested.

**Review by jhennawi — APPROVED** [review]
Looks good. Can you run a test on at least one setup in the dev suite before merging please?

**Comment by profxj** [comment]
merging and running full tests on `develop`

**Pull request merged**